databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -505,34 +509,18 @@ impl proto::val_server::Val for broker::DataBroker {
         }
     }
 
-    // Publish a signal value. Used for low frequency signals (e.g. attributes).
-    /// # Arguments
-    ///
-    ///      ```
-    ///     `request`:
-    ///      PublishValueRequest {
-    ///          signal_id: <String or i32>
-    ///          datapoint: Datapoint
-    ///      }
-    ///
-    /// # Response
-    ///      `response`:
-    ///      PublishValueResponse {
-    ///          error: Error
-    ///      }
-    ///      ```
-    ///
-    /// # Errors
-    ///  Returns (GRPC error code):
-    ///  NOT_FOUND if any of the signals are non-existant.
-    ///  PERMISSION_DENIED
-    ///      - if access is denied for any of the signals.
-    ///      - if the signal is already provided by another provider.
-    ///  INVALID_ARGUMENT
-    ///      - if the data type used in the request does not match
-    ///           the data type of the addressed signal
-    ///      - if the published value is not accepted,
-    ///           e.g. if sending an unsupported enum value
+    // Returns (GRPC error code):
+    //   NOT_FOUND if any of the signals are non-existant.
+    //   PERMISSION_DENIED
+    //       - if access is denied for any of the signals.
+    //   UNAUTHENTICATED if no credentials provided or credentials has expired
+    //   OUT_OF_RANGE is the provider is out of the min/max range specified
+    //   INVALID_ARGUMENT
+    //       - if the data type used in the request does not match
+    //            the data type of the addressed signal
+    //       - if the published value is not accepted,
+    //            e.g. if sending an unsupported enum value
```

Comment:
Is this allowed value check? Or does that give out of range error?

Reply:
It seems so, https://github.com/boschglobal/kuksa-databroker/blob/fix/doc_open_provider_stream/databroker/src/broker.rs#L434

The issue I see now is that we map to some different error but not to `INVALID_ARGUMENT` https://github.com/boschglobal/kuksa-databroker/blob/fix/doc_open_provider_stream/databroker/src/grpc/kuksa_val_v2/conversions.rs#L462

Reply:
Also fixed now

Reply:
Looking at our code we should maybe fix some conversions of OutOfBounds.

```
broker::UpdateError::OutOfBounds => tonic::Status::new(
                tonic::Code::OutOfRange,
                format!("Value out of bounds (id: {})", id),
            ),
```

According to https://grpc.github.io/grpc/cpp/md_doc_statuscodes.html `OutOfRange` seems rather to be intended for state-related errors like trying to read the forth item in an array but there are only 3, not for cases where we fail a static check like checking that argument has correct type and fulfills min/max/allowed.

So maybe refactor, remove OUT_OF_RANGE for at least v2 and describe all errors under "invalid argument", i.e. wrong type, min/max exceeded, allowed not fulfilled.

Reply:
Done